web/src/lib/roadmap/roadmapBuilder.ts

```diff
@@ -128,37 +131,32 @@ const findTaskInRoadmapByFilename = (
   roadmapBuilder: RoadmapBuilder,
   taskFilename: string
 ): TaskBuilder | undefined => {
-  for (const step of roadmapBuilder.steps) {
-    const found = step.tasks.find((task) => task.filename === taskFilename);
-    if (found) return found;
-  }
+  const found = roadmapBuilder.tasks.find((task) => task.filename === taskFilename);
+  if (found) return found;
 };
 
 const convertToRoadmap = async (roadmapBuilder: RoadmapBuilder): Promise<Roadmap> => {
   const roadmap = {
-    steps: await Promise.all(
-      roadmapBuilder.steps.map(async (step) => ({
-        ...step,
-        tasks: await Promise.all(
-          step.tasks.sort(orderByWeight).map((task) => fetchTaskByFilename(task.filename))
-        ),
-      }))
+    steps: await Promise.all(roadmapBuilder.steps),
+    tasks: await Promise.all(
+      roadmapBuilder.tasks.sort(orderByWeight).map((task) => fetchTaskByFilename(task.filename))
     ),
   };
 
-  const allFilenames = roadmap.steps.reduce(
-    (acc: string[], currStep: Step) => [...acc, ...currStep.tasks.map((task) => task.filename)],
-    []
+  roadmap.tasks.map(
+    (task: Task) =>
+      (task.stepNumber = (
+        findTaskInRoadmapByFilename(roadmapBuilder, task.filename) as TaskBuilder
+      ).step_number)
   );
 
+  const allFilenames = new Set(roadmap.tasks.map((task) => task.filename));
+
   return {
     ...roadmap,
-    steps: roadmap.steps.map((step) => ({
-      ...step,
-      tasks: step.tasks.map((task) => ({
-        ...task,
-        unlockedBy: task.unlockedBy.filter((it) => allFilenames.includes(it.filename)),
-      })),
+    tasks: roadmap.tasks.map((task) => ({
+      ...task,
+      unlockedBy: task.unlockedBy.filter((it) => allFilenames.has(it.filename)),
```

Comment:
got super confused by this for a second, I'm all "js doesn't have an array.has method" but realized it's because of the set. is there a reason we're using a set?